src/Chapter3.hs

```diff
@@ -479,6 +523,36 @@ After defining the city, implement the following functions:
    complicated task, walls can be built only if the city has a castle
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
+type People = Int
+
+data Castle = Castle String | NoCastle deriving (Show)
+data Wall = Wall | NoWall deriving (Show)
+data SpecBuilding = Church | Library deriving (Show)
+data House = House People deriving (Show)
```

Comment:
Nice type, however, it allows having houses of any number of people inside (even negative) 🧑🏼‍🤝‍🧑🏼 

To avoid building houses with more than 4 people you need to restrict usage of the Constructor and allow to create `House`s only with your `mkHouse` function. It is usually called "smart constructor". 
You can read a bit more on the pattern in here: https://kowainik.github.io/posts/haskell-mini-patterns#smart-constructor

Generally, in here, you could restrict that on type level, by using something else than `Int` 🙂

Reply:
Ok, if I get you right, smart constructors works only if I place newtype and constructor/deconstructor in separate module to prevent scope visibility of first. To cope with this problem there I'm using data class for people instead of int.